File: backend/app/schemas/payday.py
"""
发薪日配置 - 请求/响应模型
"""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from app.models.payday import PaydayConfig


class PaydayConfigBase(BaseModel):
    job_name: str = Field(..., max_length=50)
    payday: int = Field(..., ge=1, le=31)
    calendar_type: str = Field(default="solar", pattern="^(solar|lunar)$")
    estimated_salary: Optional[int] = Field(None, ge=0, description="预估工资（分）")
    is_active: int = Field(default=1, ge=0, le=1)


class PaydayConfigCreate(PaydayConfigBase):
    pass


class PaydayConfigUpdate(BaseModel):
    job_name: Optional[str] = Field(None, max_length=50)
    payday: Optional[int] = Field(None, ge=1, le=31)
    calendar_type: Optional[str] = Field(None, pattern="^(solar|lunar)$")
    estimated_salary: Optional[int] = None
    is_active: Optional[int] = Field(None, ge=0, le=1)


class PaydayConfigResponse(BaseModel):
    id: str
    user_id: str
    job_name: str
    payday: int
    calendar_type: str
    estimated_salary: Optional[int] = None
    is_active: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


def payday_to_response(m: PaydayConfig) -> dict:
    return PaydayConfigResponse.model_validate(m).model_dump()

File: miniapp/src/api/payday.ts
/**
 * 发薪日配置 - 与 backend app.api.v1.payday 一致
 */
import request from '@/utils/request'

const PREFIX = '/api/v1/payday'

/** 发薪日配置（与 PaydayConfigResponse 一致） */
export interface PaydayConfig {
  id: string
  user_id: string
  job_name: string
  payday: number
  calendar_type: 'solar' | 'lunar'
  estimated_salary: number | null
  is_active: number
  created_at: string
  updated_at: string
}

/** 新增参数 */
export interface PaydayConfigCreate {
  job_name: string
  payday: number
  calendar_type?: 'solar' | 'lunar'
  estimated_salary?: number | null
  is_active?: number
}

/** 更新参数（全部可选） */
export interface PaydayConfigUpdate {
  job_name?: string
  payday?: number
  calendar_type?: 'solar' | 'lunar'
  estimated_salary?: number | null
  is_active?: number
}

/** 列表 */
export function listPayday() {
  return request<PaydayConfig[]>({ url: PREFIX, method: 'GET' })
}

/** 新增 */
export function createPayday(data: PaydayConfigCreate) {
  return request<PaydayConfig>({ url: PREFIX, method: 'POST', data })
}

/** 单条 */
export function getPayday(configId: string) {
  return request<PaydayConfig>({ url: `${PREFIX}/${configId}`, method: 'GET' })
}

/** 更新 */
export function updatePayday(configId: string, data: PaydayConfigUpdate) {
  return request<PaydayConfig>({ url: `${PREFIX}/${configId}`, method: 'PUT', data })
}

/** 删除（204 无 body） */
export function deletePayday(configId: string) {
  return request<void>({ url: `${PREFIX}/${configId}`, method: 'DELETE' })
}
